### ISSUE TITLE
Maintenance: Add test compilation check to package builds

### ISSUE THREAD

**Issue by svozza** [aws-powertools/powertools-lambda-typescript#5073]
### Summary

Package builds (e.g. `npm run build`) only compile source code under `src/`. Test files under `tests/` are never type-checked as part of the build, meaning type errors in tests can go unnoticed until runtime.

Every package already has a `tests/tsconfig.json` configured with `noEmit: true` that includes both source and test files — it just isn't invoked during the build.

### Why is this needed?

Type errors in test files can silently accumulate and only surface at runtime. Including test compilation in the build ensures type errors are caught early and consistently.

### Which area does this relate to?

All packages

### Solution

For each package, make two changes:

**1. Remove redundant source inclusion from `tests/tsconfig.json`**

Source files imported by tests are already resolved via module resolution, so explicitly including `../src/**/*` just duplicates work that the main build already does.

For packages that include `../../testing/src/setupEnv.ts` (e.g. event-handler, logger, parameters):

```diff
-"include": ["../../testing/src/setupEnv.ts", "../src/**/*", "./**/*"]
+"include": ["../../testing/src/setupEnv.ts", "./**/*"]
```

For all other packages:

```diff
-"include": ["../src/**/*", "./**/*"]
+"include": ["./**/*"]
```

**2. Add test type-checking to the build in `package.json`**

The test type-check runs in parallel with the ESM and CJS builds (note the single `&`).

```diff
  "scripts": {
    ...
+   "build:tests": "tsc --noEmit -p tests/tsconfig.json",
-   "build": "npm run build:esm & npm run build:cjs",
+   "build": "npm run build:esm & npm run build:cjs & npm run build:tests",
    ...
  }
```

Apply to all packages.

**3. Create missing `tests/tsconfig.json` for `jmespath`**

The `jmespath` package is the only one missing this file. Create `packages/jmespath/tests/tsconfig.json`:

```json
{
  "extends": "../tsconfig.json",
  "compilerOptions": {
    "rootDir": "../",
    "noEmit": true
  },
  "include": ["./**/*"]
}
```

### Acknowledgment

- [x] This request meets [Powertools for AWS Lambda (TypeScript) Tenets](https://docs.powertools.aws.dev/lambda/typescript/latest/#tenets)
- [ ] Should this be considered in other Powertools for AWS Lambda languages? i.e. [Python](https://github.com/aws-powertools/powertools-lambda-python/), [Java](https://github.com/aws-powertools/powertools-lambda-java/), and [.NET](https://github.com/aws-powertools/powertools-lambda-dotnet/)

### Future readers

Please react with 👍 and your use case to help us understand customer demand.

**Comment by dreamorosi** [comment]
I think in the age of agents this is very much needed so that we can check their work.

My only caveat - which seems to be addressed by the proposal - is to make sure that we don't introduce the tests in the built bundle that ends up on npm, which was the original reason why we didn't add them to the build.
